core/stream/convert.go

```diff
@@ -364,60 +372,181 @@ func writeUintClamped(bs *binary.BitStream, bits uint64, count uint32) {
 	bs.Write(u64.Min(bits, limit), count)
 }
 
+func ftouMinMax(count int, dst, src buf, min, max *float64) error {
+	srcTy := src.component.DataType
+	srcIsF16, srcIsF32, srcIsF64 := srcTy.Is(F16), srcTy.Is(F32), srcTy.Is(F64)
+	sourceStream := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
+	switch {
+	case srcIsF16:
+		for i := 0; i < count; i++ {
+			f := f16.Number(sourceStream.Read(16)).Float32()
+			if !math.IsInf(float64(f), 0) {
+				if float64(f) < *min {
+					*min = float64(f)
+				}
+				if float64(f) > *max {
+					*max = float64(f)
+				}
+			}
+			sourceStream.ReadPos += src.stride - 16
+		}
+	case srcIsF32:
+		for i := 0; i < count; i++ {
+			f := math.Float32frombits(uint32(sourceStream.Read(32)))
+			if !math.IsInf(float64(f), 0) {
+				if float64(f) < *min {
+					*min = float64(f)
+				}
+				if float64(f) > *max {
+					*max = float64(f)
+				}
+			}
+			sourceStream.ReadPos += src.stride - 32
+		}
+	case srcIsF64:
+		for i := 0; i < count; i++ {
+			f := math.Float64frombits(sourceStream.Read(64))
+			if !math.IsInf(f, 0) {
+				if f < *min {
+					*min = f
+				}
+				if f > *max {
+					*max = f
+				}
+			}
+			sourceStream.ReadPos += src.stride - 64
+		}
+	default:
+		for i := 0; i < count; i++ {
+			f := math.Float64frombits(sourceStream.Read(64))
+			if !math.IsInf(f, 0) {
+				if f < *min {
+					*min = f
+				}
+				if f > *max {
+					*max = f
+				}
+			}
+			sourceStream.ReadPos += src.stride - 64
+		}
+	}
+	return nil
+}
+
 // float to unsigned int
-func ftou(count int, dst, src buf) error {
+func ftou(count int, dst, src buf, min, max float64) error {
 	dstTy, srcTy := dst.component.DataType, src.component.DataType
 	srcIsF16, srcIsF32, srcIsF64 := srcTy.Is(F16), srcTy.Is(F32), srcTy.Is(F64)
 	srcExpBits, srcManBits := srcTy.GetFloat().ExponentBits, srcTy.GetFloat().MantissaBits
 	norm := dst.component.IsNormalized()
 	dstBits, srcBits := dstTy.Bits(), srcTy.Bits()
 	dstMask := (1 << dstBits) - 1
-	ss := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
-	ds := binary.BitStream{Data: dst.bytes, WritePos: dst.offset}
+	sourceStream := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
+	destStream := binary.BitStream{Data: dst.bytes, WritePos: dst.offset}
 	switch {
 	case srcIsF16:
 		scale := float32(dstMask)
+		sourceStream.ReadPos = src.offset
 		for i := 0; i < count; i++ {
-			f := f16.Number(ss.Read(16)).Float32()
+			f := f16.Number(sourceStream.Read(16)).Float32()
+			fMin := float32(min)
+			fMax := float32(max)
 			if norm {
+				if src.component.Channel == Channel_Depth {
+					if fMax != fMin {
+						f = (f - fMin) * (1 / (fMax - fMin))
+					} else {
+						f = 0
+					}
```

Comment:
This could be done out of the loop for a bit better performance. I'm fine leaving the way it is, though, if makes the code look too complicated.